pkg/services/accesscontrol/authorizer.go

```diff
@@ -0,0 +1,131 @@
+package accesscontrol
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/authlib/claims"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+)
+
+// ResourceResolver is called before authorization is performed.
+// It can be used to translate resoruce name into one or more valid scopes that
+// will be used for authorization. If more than one scopes are returned from a resolver
+// only one needs to match to allow call to be authorized.
+type ResourceResolver interface {
+	Resolve(ctx context.Context, ns claims.NamespaceInfo, name string) ([]string, error)
+}
+
+// ResourceResolverFunc is an adapter so that functions can implement ResourceResolver.
+type ResourceResolverFunc func(ctx context.Context, ns claims.NamespaceInfo, name string) ([]string, error)
+
+func (r ResourceResolverFunc) Resolve(ctx context.Context, ns claims.NamespaceInfo, name string) ([]string, error) {
+	return r(ctx, ns, name)
+}
+
+type ResourceAuthorizerOptions struct {
+	// Resource is the resource name in plural.
+	Resource string
+	// Attr is attribute used for resource scope. It's usually 'id' or 'uid'
+	// depending on what is stored for the resource.
+	Attr string
+	// Mapping is used to translate k8s verb to rbac action.
+	// Key is the desired verb and value the rbac action it should be translated into.
+	Mapping map[string]string
+	// Resolver if passed can translate into one or more scopes used to authorize resource.
+	// This is useful when stored scopes are based on something else than k8s name or
+	// for resources that inherit permission from folder.
+	Resolver ResourceResolver
+}
+
+var _ claims.AccessClient = (*LegacyAccessClient)(nil)
+
+func NewLegacyAccessClient(ac AccessControl, opts ...ResourceAuthorizerOptions) *LegacyAccessClient {
```

Comment:
What's the intent with `opts ...ResourceAuthorizeOptions` ? Having a single `LegacyAccessClient` for multiple resource kind?
Asking because we seem to currently instantiate on a kind basis, so not sure that's needed and maybe we could simplify.

Reply:
The current pr is only adding checks for users but it is supported for several resources. LegacyAccessClient will load options for a resource and perform evaluation from there.